### PR TITLE
detect/transform: fix leak in xor transform parse

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/7229 follow up

Describe changes:
- detect/transform: fix leak in xor transform parse

Fixes: 8984bc680112 ("transforms: move xor to rust")

### CODE REVIEW

File: rust/src/detect/transforms/xor.rs [hunk 1]
```diff
@@ -60,13 +60,17 @@ unsafe fn xor_parse(raw: *const std::os::raw::c_char) -> *mut c_void {
...
+    de: *mut c_void, s: *mut c_void, opt_str: *const std::os::raw::c_char,
 ) -> c_int {
     let ctx = xor_parse(opt_str);
     if ctx.is_null() {
         return -1;
     }
-    return DetectSignatureAddTransform(s, G_TRANSFORM_XOR_ID, ctx);
+    let r = DetectSignatureAddTransform(s, G_TRANSFORM_XOR_ID, ctx);
+    if r != 0 {
+        xor_free(de, ctx);
```
Comment (catenacyber) [inline rust/src/detect/transforms/xor.rs:71]: Happens if you do not have content to apply the transform to for example

**Comment by catenacyber** [comment]
I wonder why oss-fuzz did not catch it

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 23514

File: rust/src/detect/transforms/xor.rs (hunk 1 above)
Comment (inashivb) [reply rust/src/detect/transforms/xor.rs:71]: For me even ASAN does not flag it. I cannot reproduce it if I do not have a content to apply to the transform to..
How did you find it?

**Review by victorjulien — APPROVED** [review]

File: rust/src/detect/transforms/xor.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/detect/transforms/xor.rs:71]: Maybe you need to enable it with export `ASAN_OPTIONS="detect_leaks=1"` ?

I got it with something like `alert ip any any -> any any ( xor: "12"; sid:1)` where there is no buffer to do the transform on

**Pull request merged**